src/Arachne.h

```diff
@@ -615,7 +615,7 @@ createThreadWithClass(int threadClass, _Callable&& __f, _Args&&... __args) {
     // the one with the fewest Arachne threads.
     uint32_t kId;
     CoreList* coreList = coreManager->getCores(threadClass);
-    if (coreList == NULL)
+    if ((coreList == NULL) || (coreList->size() == 0))
```

Comment:
Maybe this check belongs in the DefaultCoreManager; that way CoreManagers which do not allow empty core lists to be returned don't have to pay for the branch.